modules/ROOT/pages/mule-object-stores.adoc

```diff
@@ -0,0 +1,89 @@
+= Store Application Data Using Object Stores
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:keywords: mule, object, store
+
+An object store is a facility for storing objects in or across Mule applications. Mule runtime engine(Mule) uses object stores to persist data for eventual retrieval. Internally, Mule uses object stores in various filters, routers, and other message processors that need to store states between messages.
+
+If you deploy your application to CloudHub, you can also use Object Store V2.
+
+== Use Cases
+
+You can configure a custom object store to use in the following scenarios:
+
+* When configuring an Idempotent Message Validator, to store the unique IDs in the specified object store.
+* When configuring a Redelivery policy in a listener(HTTP, File, JMS, VM, Socket), to store the redelivery count in the specified object store.
+* When configuring a Cache Scope that uses a custom caching strategy, to store the cached data in the specified object store.
+* When a custom component uses an object store to persist information.
+* When storing or retrieving information from a Mule flow, by using the Object Store Connector.
+
+== Configuration
+
+By default, each Mule application has an object store that is persistent and is always available to the app. You can also create custom object stores, which can be persistent or in-memory.
+
+You can configure a custom object store for a specific component that supports its usage, or configure the object store as a global element, to be accessible for any component in your app.
+
+=== Configure a Global Object Store
+
+To configure a global object store that can be referenced by other components in your application, follow these steps in Anypoint Studio:
+
+. Open the *Global Elements* tab.
+. Click *Create*.
+. Select *Component configurations* > *Object Store*.
+. Complete the required configuration attributes and click *OK*. +
+image:mruntime-object-store-global.png[global object store]
+
+Alternatively, you can create a global object store by including the following code as a top-level element in your application XML:
+
+[source,xml,linenums]
+----
+<os:object-store name="myObjectStore"
+   entryTtl="1"
+   entryTtlUnit="HOURS"
+   maxEntries="100"
+   persistent="true"
+   expirationInterval="30"
+   expirationIntervalUnit="MINUTES" />
+----
+
+Configure each attribute with the desired value. +
+See xref:object-store-connector::object-store-connector-reference.adoc#objectStore[Object Store Reference] for a definition of each XML attribute.
+
+== Configure an Object Store for a Specific Component
+
+You can configure an object store that is available only for a specific component. For example, to configure an object store for an Idempotent Message Validator, follow these steps in Anypoint Studio:
```

Comment:
Would you also confirm the capitalization of Idempotent Message Validator?
If not specific to MuleSoft, I suggest lowercase.

Reply:
https://docs.mulesoft.com/mule-runtime/4.3/idempotent-message-validator "Idempontent Message Validator" component.